server/blob/blob-api/src/test/java/org/apache/james/blob/api/BucketBlobStoreContract.java

```diff
@@ -0,0 +1,154 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+import org.junit.jupiter.api.Test;
+
+public interface BucketBlobStoreContract extends MetricableBlobStoreContract {
```

Comment:
I prefer to put these tests into BucketStoreContract because all BlobStore impls should have the Bucket capability feature, then it should be in the top contract.

Reply:
I don't see any test about saving blobs in different buckets

Reply:
Don't get your first comment, this is a new contract interface with the bucket tests inside already...?

Reply:
@trantienduchn How about having the blobStore contract extending the `BucketBlobStoreContract`? Then there is no issue doing that later, when all implementation supports this.

Reply:
Isolated the contract and added the missing test